src/genegraph/annotate.clj

```diff
@@ -201,3 +202,16 @@
 (def add-jsonld-interceptor
   {:name ::add-jsonld-interceptor
    :enter (fn [e] (xform-types/add-model-jsonld e))})
+
+(def add-graphql-serialization-interceptor
+  "Interceptor for attaching graphql query information for an event"
+  {:name ::add-graphql-serialization-interceptor
+   :enter (fn [e] (let [e-with-params (xform-types/add-event-graphql e)
+                        gql-params (:graphql-params e-with-params)]
+                    (if (not-empty gql-params)
+                      (assoc e-with-params
+                        ::graphql-serialization
+                        (experimental-schema/query
+                          (:query gql-params)
+                          (:variables gql-params)))
+                      e)))})
```

Comment:
I think this code deserves its own file, rather than piling into genegraph.annotate. If nothing else, I would like to isolate the GraphQL dependency a little better.